nautobot/extras/models/metadata.py

```diff
@@ -135,3 +138,70 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         # TODO: should we behave like CustomFieldChoice and raise a ProtectedError if this choice is in use anywhere?
         super().delete(*args, **kwargs)
+
+
+class ObjectMetadata(BaseModel, ChangeLoggedModel):
+    metadata_type = models.ForeignKey(
+        to=MetadataType,
+        on_delete=models.CASCADE,
+        related_name="object_metadatas",
+    )
+    contact = models.ForeignKey(
+        to=Contact,
+        on_delete=models.SET_NULL,
+        related_name="object_metadatas",
+        null=True,
+    )
+    team = models.ForeignKey(
+        to=Team,
+        on_delete=models.SET_NULL,
```

Comment:
These should either be CASCADE or PROTECT as well, shouldn't they?

Reply:
I think I am fine with either `SET_NULL` or `PROTECT`. I am not sure if deleting a contact should delete the `ObjectMetaData` instances as well.

Reply:
SET_NULL means that deleting a contact results in an ObjectMetadata of type Contact/Team that doesn't have a Contact or Team set - that should be invalid data.